example/src/main/java/com/basistheory/android/example/viewmodel/ApiViewModel.kt

```diff
@@ -0,0 +1,90 @@
+package com.basistheory.android.example.viewmodel
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.liveData
+import com.basistheory.android.example.BuildConfig
+import com.basistheory.android.example.R
+import com.basistheory.android.example.util.prettyPrintJson
+import com.basistheory.android.service.BasisTheoryElements
+import com.basistheory.android.service.ProxyRequest
+
+open class ApiViewModel(application: Application): AndroidViewModel(application) {
+    private val _errorMessage = MutableLiveData<String?>(null)
+    val errorMessage: LiveData<String?>
+        get() = _errorMessage
+
+    private val _result = MutableLiveData<String?>(null)
+    val result: LiveData<String?>
+        get() = _result
+
+    private val _proxyResult = MutableLiveData<String?>(null)
+    val proxyResult: LiveData<String?>
+        get() = _proxyResult
+
+    private val bt = BasisTheoryElements.builder()
+        .apiUrl(BuildConfig.BASIS_THEORY_API_URL)
+        .apiKey(BuildConfig.BASIS_THEORY_API_KEY)
+        .build()
+
+    fun tokenize(payload: Any): LiveData<Any> = liveData {
+        _errorMessage.value = null
+        _result.value = null
+
+        runCatching {
+            bt.tokenize(payload)
+        }.fold(
+            onSuccess = {
+                _result.value = it.prettyPrintJson()
+                emit(it)
+            },
+            onFailure = {
+                _errorMessage.value = getApplication<Application>()
+                    .resources
+                    .getString(R.string.tokenize_error, it)
+            }
+        )
+    }
+
+    fun proxy(proxyRequest: ProxyRequest): LiveData<Any> = liveData {
+        _errorMessage.value = null
+        _proxyResult.value = null
+
+        runCatching {
+            bt.proxy.post(proxyRequest)
+        }.fold(
+            onSuccess = {
+                _proxyResult.value = it?.prettyPrintJson()
+                if (it != null) {
+                    emit(it)
+                }
+            },
+            onFailure = {
+                _errorMessage.value = getApplication<Application>()
+                    .resources
+                    .getString(R.string.proxy_error, it)
+            }
+        )
+    }
+
+    fun getToken(id: String): LiveData<Any> = liveData {
```

Comment:
nitpick: Can this return `LiveData<Token>` to avoid needing to cast in the fragment? Referring to this:
```kotlin
(it as Token).data.getElementValueReference("number")
```